frontend/app/api.ts

```diff
@@ -0,0 +1,67 @@
+import Cookies from 'js-cookie';
+
+import { Tool } from './interfaces';
+
+export const getCSRFToken = (): string | undefined => Cookies.get('csrftoken');
+
+const API_BASE = '/api';
+const JSON_MIME_TYPE = 'application/json';
+
+const BASE_MUTATION_HEADERS: HeadersInit = {
+  Accept: JSON_MIME_TYPE,
+  'Content-Type': JSON_MIME_TYPE,
+  'X-Requested-With': 'XMLHttpRequest'
+};
+
+const createErrorMessage = async (res: Response): Promise<string> => {
+  let errorBody;
+  try {
+    errorBody = await res.json();
+  } catch {
+    console.error('Error body was not JSON.');
+    errorBody = undefined;
+  }
+  return (
+    'Error occurred! ' +
+    `Status: ${res.status}` + (res.statusText !== '' ? ` (${res.statusText})` : '') +
+    (errorBody !== undefined ? '; Body: ' + JSON.stringify(errorBody) : '.')
+  );
+};
+
+async function getTools (): Promise<Tool[]> {
+  const url = `${API_BASE}/lti_tools/`;
+  const res = await fetch(url);
+  if (!res.ok) {
+    console.error(res);
+    throw new Error(await createErrorMessage(res));
+  }
+  const data: Tool[] = await res.json();
+  return data;
+}
+
+interface UpdateToolNavData {
+  canvasToolId: number
+  navEnabled: boolean
+}
+
+async function updateToolNav (data: UpdateToolNavData): Promise<void> {
+  const { canvasToolId, navEnabled } = data;
+  const body = { navigation_enabled: navEnabled };
+  const url = `${API_BASE}/lti_tools/${canvasToolId}/`;
+  const requestInit: RequestInit = {
+    method: 'PUT',
+    body: JSON.stringify(body),
+    headers: {
+      ...BASE_MUTATION_HEADERS,
+      'X-CSRFTOKEN': getCSRFToken() ?? ''
```

Comment:
We could maybe throw an error in `getCSRFToken` instead, but it seems unlikely that the CSRF token won't be there.